skimage/transform/_warps_cy.pyx

```diff
@@ -119,8 +120,13 @@ def _warp_fast(cnp.ndarray image, cnp.ndarray H, output_shape=None,
 
     """
 
-    cdef double[:, ::1] img = np.ascontiguousarray(image, dtype=np.double)
-    cdef double[:, ::1] M = np.ascontiguousarray(H)
+    cdef floating[:, ::1] img = np.ascontiguousarray(image)
+    cdef floating[:, ::1] M = np.ascontiguousarray(H)
+
+    if floating is float:
```

Comment:
Is float ambiguous here? I'm concerned there is a difference between python floats and c floats.

Reply:
In Cython context, I think that `float` belongs to `cython.float`. So I suppose that it is not ambiguous...

Reply:
I took it from [cython doc](http://docs.cython.org/en/latest/src/userguide/fusedtypes.html)  in **Type Checking Specializations** section ;-).

Reply:
The fact that you used the word "think" and "not ambiguous" in the same sentence to me makes it ambiguous.

The challenge comes when having code like this:

![image](https://user-images.githubusercontent.com/90008/63992156-9a1e9700-cab8-11e9-84ac-3afd8d3a13ae.png)

`float` clearly means different things when used in python+numpy, or in cython.

I would just replace `float` with `np.float32_t`, making it explicit that you mean float32.

Reply:
OK, I was thinking that you where talking about _computer ambiguity_, not _understanding ambiguity_. Sorry for my poor English speaking/understanding.
On python side, the ambiguity is clearly there, but in cython context I don't think so, the whole test block is explicit:
```python
    if floating is float:
        dtype = np.float32
    else:
        dtype = np.double
```
We can make it even more explicit using:
```python
    if floating is float:
        dtype = np.float32
    elif floating is double:
        dtype = np.double
    else:
        raise TypeError("Only float32 and float64 arrays are supported.")
```

Reply:
The last part of my answer is wrong, since floating can only be `float` or `double`. Sorry :frowning_face: